lib/matplotlib/ticker.py

```diff
@@ -1882,11 +1889,17 @@ def set_params(self, **kwargs):
             self._extended_steps = self._staircase(self._steps)
         if 'integer' in kwargs:
             self._integer = kwargs['integer']
+        if 'tick_space_heuristic' in kwargs:
+            tick_space_heuristic = kwargs['tick_space_heuristic']
```

Comment:
I think we just use `kwarg.pop` these days...

Reply:
That's fine by me. I was just following the style used in the rest of the `set_params` function. Do you think I should convert the entire function then? Seems like something for a different PR, but I don't mind doing it here.